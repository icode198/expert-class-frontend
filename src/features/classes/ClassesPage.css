* {
  text-decoration: none;
}

.classes-container {
  width: 1070px;
  height: 100vh;
  padding: 20px;
  margin-left: 240px;
}

.heading {
  text-align: center;
  margin: 25px auto;
  text-transform: uppercase;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.carousel-container {
  height: 100%;
}

.carousel-component {
  height: 100%;
}

.classes-item {
  width: 300px;
  border-radius: 15px;
  background-color: #97bf0f;
  color: #fff;
}

.text-container,
.image-container {
  margin: 0 auto;
  text-align: center;
}

.text-container {
  padding: 20px;
}

.image {
  margin: 0 auto;
  width: 100%;
  height: 300px;
}

.classes-title {
  text-transform: uppercase;
}

.class-instructor {
  padding: 0.5rem;
}

.classes-social-btns {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 1rem;
}

.classes-social-icon-circle {
  width: 26px;
  height: 26px;
  padding: 4px;
  margin: 0 5px;
  border: 2px solid lightgreen;
  border-radius: 50%;
  cursor: pointer;
}

.classes-social-icon {
  color: lightgreen;
  fill: lightgreen;
}
