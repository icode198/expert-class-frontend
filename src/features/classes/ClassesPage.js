import './ClassesPage.css';
import '@brainhubeu/react-carousel/lib/style.css';
import Carousel, { slidesToShowPlugin, autoplayPlugin } from '@brainhubeu/react-carousel';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { fetchClassesData } from './classesSlice';
import { reloadClasses } from '../removeClass/removeClassSlice';
import SocialIcons from './socialIcons';

export default function ClassesPage() {
  const dispatch = useDispatch();

  const classes = useSelector((state) => state.classes.classes);

  useEffect(() => {
    dispatch(fetchClassesData());
    dispatch(reloadClasses());
  }, []);

  const classesList = classes.map((c) => (
    <NavLink to={`/class/${c.id}`} key={c.id} className="classes-item">
      <div className="image-container">
        <img className="image" src={c.course_image_url} alt="course" />
      </div>
      <div className="text-container">
        <h3 className="classes-title">{c.title}</h3>
        <p className="class-instructor">{c.instructor}</p>
        <SocialIcons />
      </div>
    </NavLink>
  ));

  return (
    <div className="classes-container">
      <h2 className="heading">Classes</h2>
      <Carousel
        plugins={[
          'centered',
          'infinite',
          'arrows',
          {
            resolve: slidesToShowPlugin,
            options: {
              numberOfSlides: 3,
            },
          },
          {
            resolve: autoplayPlugin,
            options: {
              interval: 3000,
            },
          },
        ]}
        animationSpeed={1000}
        className="carousel-component"
      >
        {classesList}
      </Carousel>
    </div>
  );
}
