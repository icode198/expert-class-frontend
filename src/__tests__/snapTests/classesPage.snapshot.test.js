import React from 'react';
import { cleanup } from '@testing-library/react';
import { renderWithRedux } from '../../tests/test-utils';
import ClassesPage from '../../pages/classesPage/classesPage';

afterEach(cleanup);

describe('Classes Page', () => {
  test('Renders the page', () => {
    const app = renderWithRedux(<ClassesPage />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
