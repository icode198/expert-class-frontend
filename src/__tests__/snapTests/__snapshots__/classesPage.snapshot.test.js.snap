// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Classes Page Renders the page 1`] = `null`;
